terraform/ecs/ecs.tf

```diff
@@ -0,0 +1,88 @@
+resource "aws_ecs_cluster" "ecscluster" {
+  name = "${var.projectname}-cluster"
+}
+
+resource "aws_security_group" "ecs-ec2-sg" {
+    name   = "${var.projectname}-ecs-ec2-sg"
+    vpc_id = var.vpc_id
+
+    lifecycle {
+        create_before_destroy = true
+    }
+
+    ingress {
+        from_port       = 0
+        to_port         = 65535 
+        protocol        = "tcp" 
+        cidr_blocks     = ["0.0.0.0/0"]
+        security_groups = [var.alb_sg] 
+        description     = "From ALB"
+    }
+}
+
+resource "aws_launch_configuration" "lc" {
+  name = "${var.projectname}-lc"
+  image_id = var.ami_id
+  instance_type = var.inst_type
+  key_name      = var.inst_key != "" ? var.inst_key : ""
+  security_groups = [aws_security_group.ecs-ec2-sg.id]
+  user_data = base64encode(data.template_file.userdata.rendered)
+  iam_instance_profile = aws_iam_instance_profile.instprof.name
+  lifecycle {
+    create_before_destroy = true
+  }
+
+}
+
+resource "aws_autoscaling_group" "asg" {
+  availability_zones = var.az
+  desired_capacity   = var.asg_desired_capac
+  max_size           = var.asg_max_size
+  min_size           = var.asg_min_size
+  launch_configuration = aws_launch_configuration.lc.name
+  vpc_zone_identifier = [var.pub_sub1,var.pub_sub2]
+}
+
+data "template_file" "userdata" {
+  template = file("${path.module}/userdata.tpl")
+  vars = {
+    ecs_cluster_name = aws_ecs_cluster.ecscluster.name
+    efs_id = var.efs_id
+  }
+}
+
+data "aws_iam_role" "ecs_role" {
+  name = "ecsTaskExecutionRole"
```

Comment:
as explained in slack, we need this as a separate role.. @CarineOliveira will fix on her code